src/main/java/edit_pending_list/EditPendingListInteractor.java

```diff
@@ -0,0 +1,69 @@
+package edit_pending_list;
+
+import Entities.Group;
+import Entities.User;
+
+import java.util.HashMap;
+
+/**
+ * The edit pending list use case.
+ * Allows group leaders to accept or reject users who've applied to their group.
+ */
+
+public class EditPendingListInteractor implements EditPendingListInputBoundary {
+
+    final EditPendingListDsGateway dsGateway;
+    final EditPendingListOutputBoundary presenter;
+
+    /**
+     * @param dsGateway the data access interface
+     * @param presenter the output boundary implemented by EditPendingListPresenter
+     */
+    public EditPendingListInteractor(EditPendingListDsGateway dsGateway, EditPendingListOutputBoundary presenter) {
+        this.dsGateway = dsGateway;
+        this.presenter = presenter;
+    }
+
+    /**
+     * Adds the user and group to the appropriate lists based on whether the user was accepted or rejected.
+     * @param requestModel the requestModel for the edit pending list use case
+     * @return a response model
+     */
+    @Override
+    public EditPendingListResponseModel addOrRemoveUser(EditPendingListRequestModel requestModel) {
+        String username = requestModel.getUsername();
+        String groupName = requestModel.getGroupName();
+        boolean pendingStatus = requestModel.getPendingStatus();
+
+        if (!dsGateway.userIdentifierExists(username)) {
+            return presenter.prepareFailView("This user has deleted their account.");
+        }
+
+        User user = dsGateway.getUser(username);
+        Group group = dsGateway.getGroup(groupName);
+        HashMap<String, User> userMap = dsGateway.loadUsers();
```

Comment:
Nice to see your usage of the userMap -- this is something I was a bit confused about in my use cases and I'll change my code to match yours (I did it in a more convoluted way 👎 )